devices/usbc-cdc/usbphy-imxrt117x.c

```diff
@@ -0,0 +1,154 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * i.MX RT117x USB physical layer controller
+ *
+ * Copyright 2021 Phoenix Systems
+ * Author: Gerard Swiderski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#include "client.h"
+#include "usbphy.h"
+
+
+/* Size of memory pool aligned to USB_BUFFER_SIZE, used by USB descriptors */
+#define USB_POOL_SIZE ((ENDPOINTS_DIR_NB * ENDPOINTS_NUMBER + 1) * USB_BUFFER_SIZE)
+
+
+static struct {
+	unsigned char pool[USB_POOL_SIZE] __attribute__((aligned(USB_BUFFER_SIZE)));
+	size_t used;
+} phyusb_common;
+
+
+enum {
+	/* Power-Down Register */
+	usbphy_pwd, usbphy_pwd_set, usbphy_pwd_clr, usbphy_pwd_tog,
+	/* Transmitter & Receiver Control Registers */
+	usbphy_tx, usbphy_tx_set, usbphy_tx_clr, usbphy_tx_tog,
+	usbphy_rx, usbphy_rx_set, usbphy_rx_clr, usbphy_rx_tog,
+	/* General Control Register */
+	usbphy_ctrl, usbphy_ctrl_set, usbphy_ctrl_clr, usbphy_ctrl_tog,
+	/* USB Status & Debug Registers */
+	usbphy_status,
+	usbphy_debug = 20, usbphy_debug_set, usbphy_debug_clr, usbphy_debug_tog,
+	/* UTMI Status & Debug Registers */
+	usbphy_debug0_status, usbphy_debug1 = 28,
+	usbphy_debug1_set, usbphy_debug1_clr, usbphy_debug1_tog,
+	/* UTMI RTL */
+	usbphy_version,
+	/* PLL Control & Status */
+	usbphy_pll_sic = 40, usbphy_pll_sic_set, usbphy_pll_sic_clr, usbphy_pll_sic_tog,
+	/* VBUS Detector Control */
+	usbphy_vbus_detect = 48,
+	usbphy_vbus_detect_set, usbphy_vbus_detect_clr, usbphy_vbus_detect_tog, usbphy_vbus_detect_stat,
+	/* Charger Detect Control */
+	usbphy_chrg_detect = 56,
+	usbphy_chrg_detect_set, usbphy_chrg_detect_clr, usbphy_chrg_detect_tog, usbphy_chrg_detect_stat,
+	/* Analog Control Register */
+	usbphy_anactrl = 64,
+	usbphy_anactrl_set, usbphy_anactrl_clr, usbphy_anactrl_tog,
+	/* Loopback Control */
+	usbphy_loopback, usbphy_loopback_set, usbphy_loopback_clr, usbphy_loopback_tog,
+	/* Loopback Packet Number Select Register */
+	usbphy_loopback_hsfscnt, usbphy_loopback_hsfscnt_set, usbphy_loopback_hsfscnt_clr, usbphy_loopback_hsfscnt_tog,
+	/* Trim Override Enable */
+	usbphy_trim_override_en, usbphy_trim_override_en_set, usbphy_trim_override_en_clr, usbphy_trim_override_en_tog,
+};
+
+
+void *usbclient_allocBuff(u32 size)
+{
+	size_t org = USB_BUFFER_SIZE * phyusb_common.used;
+
+	if ((size % USB_BUFFER_SIZE) != 0 || org + size > USB_POOL_SIZE)
+		return NULL;
+
+	phyusb_common.used += size / USB_BUFFER_SIZE;
+
+	return &phyusb_common.pool[org];
+}
+
+
+void usbclient_buffReset(void)
+{
+	phyusb_common.used = 0;
+}
+
+
+void *phy_getBase(void)
+{
+	return (void *)USB0_BASE_ADDR;
```

Comment:
Casting to `void *` is redundant, define already contains casting to `void *`

Reply:
Right, previously `#define` did not contain `(void*)`, so yse it's redundant now.

Reply:
Additionally fixed in other phy implementations as well.